cpp/src/cli.cpp

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ * Licensed under MIT, see full license in LICENSE
+ * SPDX-License-Identifier: MIT
+ * License-Filename: LICENSE
+ */
+
+#include <charconv>
+#include <iostream>
+
+#include <hf/flexpolyline.h>
+
+std::string_view remove_decoration(std::string_view x, std::string_view prefix,
+                                   std::string_view suffix) {
+  if (x.size() < suffix.size() + suffix.size() || x.substr(0, prefix.size()) != prefix ||
+      x.substr(x.size() - suffix.length(), suffix.length()) != suffix) {
+    throw std::runtime_error(std::string(prefix) + suffix.data() + " missing");
+  }
+  x.remove_prefix(prefix.size());
+  x.remove_suffix(suffix.size());
+  return x;
+}
+
+std::string_view split_next(std::string_view &x, std::string_view sep, bool required) {
+  auto pos = x.find(sep);
+  auto result = x.substr(0, pos);
+  if (pos == std::string::npos) {
+    if (required) {
+      throw std::runtime_error(std::string("Missing seperator: ") + sep.data());
+    }
+    x = std::string_view();
+  } else {
+    x = x.substr(pos + sep.size());
+  }
+  return result;
+}
+
+hf::flexpolyline::Precision parse_precision(std::string_view x) {
```

Comment:
I was also thinking that it could. The current version is not fast enough, nor does enough error checking/propagation for that, though, so maybe something left for a follow-up review?

Reply:
Yeah, passing the entire polyline to `atoi()` is not the fastest, I see that now :-D. 

IMO, we can open a ticket to resolve (or not, depending on preference) the issues I raised and merge this, as soon as you are happy with it.